lightning/src/util/events.rs

```diff
@@ -67,6 +68,38 @@ pub enum PaymentPurpose {
 	SpontaneousPayment(PaymentPreimage),
 }
 
+#[derive(Clone, Debug)]
+/// Some information provided on the closure source of the channel halting.
+pub enum ClosureReason {
+	/// Closure generated from receiving a peer error message by ChannelManager::handle_error
+	CounterpartyForceClosed {
+		/// The error is coming from the peer, there *might* be a human-readable msg
+		peer_msg: Option<String>,
+	},
+	/// Closure generated from ChannelManager::force_close_channel
```

Comment:
I think this should mention that this was the user who did this - only they can call that. Also, function links should be inside [`s